core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java

```diff
@@ -66,8 +67,13 @@ public MessageDestination getAnonymousDestination()
 
 	public Set<MessageDestination> getBuddyDestinations()
 	{
-		return buddiesAnonymized.stream().filter(ba -> ba.getSendingStatus() == Status.ACCEPTED).map(ba -> ba.getUserAnonymized())
-				.map(ua -> ua.getAnonymousDestination()).collect(Collectors.toSet());
+		return mapAvailableDestinations(buddiesAnonymized.stream()).collect(Collectors.toSet());
+	}
+
+	public static Stream<MessageDestination> mapAvailableDestinations(Stream<BuddyAnonymized> buddiesAnonymized2)
```

Comment:
A couple of comments on this method signature:
* The name does not tell what the method does.
* The parameter name with a "2" in it is not very logical and it is an indicator of something else: ``UserAnonymized`` already knows the ``buddiesAnonymized``, so passing that into it is strange.
* Passing streams through public APIs is uncommon. We commonly use collections in such a case.

My proposal is to solve this in a different way: link anonymized buddy DTOs to the anonymized user DTOs, so you don't need to go to the database to find out where to send the messages to.